File: android/app/src/main/kotlin/com/typography/animations/loop/Shake.kt
package com.typography.animations.loop

import com.typography.AnimatedTextState
import kotlin.math.*

fun evaluateShake(angle: Float): AnimatedTextState {
    val freq = 15f
    val intensity = 0.02f
    return AnimatedTextState(
        offsetX = sin(angle * freq) * intensity,
        offsetY = cos(angle * freq * 0.7f) * intensity
    )
}

File: lib/utils/animations/loop/shake.dart
import 'dart:math';
import '../base/animated_text_state.dart';

AnimatedTextState evaluateShake(double angle) {
  const freq = 15.0;
  const intensity = 0.01;
  return AnimatedTextState(
    offsetX: sin(angle * freq) * intensity,
  );
}
